src/catch2/internal/catch_istream.cpp

```diff
@@ -86,29 +96,76 @@ namespace Detail {
 
         ///////////////////////////////////////////////////////////////////////////
 
+#if defined( CATCH_PLATFORM_WINDOWS )
+        bool enableVirtualTerminalSupport( DWORD stdHandle ) {
+            HANDLE outputHandle = GetStdHandle( stdHandle );
+            DWORD mode = 0;
+            const DWORD requiredMode = ENABLE_PROCESSED_OUTPUT |
+                                ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+            if ( GetConsoleMode( outputHandle, &mode ) &&
+                    ( mode & requiredMode ) == requiredMode ) {
+                // VT100 style sequence processing has to be enabled by
+                // explicit opt-in.
+                const DWORD newMode = mode | requiredMode;
+                if( SetConsoleMode( outputHandle, newMode ) )
+                {
+                    return true;
+                }
+                // Restore fail-safe state.
+                SetConsoleMode( outputHandle, mode );
+            }
+            return false;
+        }
+#endif
+
         class CoutStream final : public IStream {
             std::ostream m_os;
+            bool m_isatty;
         public:
             // Store the streambuf from cout up-front because
             // cout may get redirected when running tests
-            CoutStream() : m_os( Catch::cout().rdbuf() ) {}
+            CoutStream() : m_os( Catch::cout().rdbuf() ) {
+                m_isatty = true;
+#if defined( CATCH_INTERNAL_HAS_ISATTY ) && \
+    !( defined( __DJGPP__ ) && defined( __STRICT_ANSI__ ) )
+                ErrnoGuard _; // for isatty
+                m_isatty = m_isatty && isatty( STDOUT_FILENO );
+#elif defined( CATCH_PLATFORM_WINDOWS )
+                m_isatty = m_isatty && enableVirtualTerminalSupport( STD_OUTPUT_HANDLE );
+#endif
+#if defined( CATCH_PLATFORM_MAC ) || defined( CATCH_PLATFORM_IPHONE )
+                    m_isatty = m_isatty && !isDebuggerActive();
+#endif
+            }
 
         public: // IStream
             std::ostream& stream() override { return m_os; }
-            bool isConsole() const override { return true; }
+            bool isConsole() const override { return m_isatty; }
```

Comment:
This does not pass the tests. Now `isConsole` reports `false` in all of the unit tests when run in the CI - but its doing so rightfully.

Reply:
Two options - injecting a proper terminal emulator for testing or changing the expected results?